File: backend/src/main/java/com/schoolfinder/domain/School.java
package com.schoolfinder.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "school")
@Getter
@Setter
@NoArgsConstructor
public class School {

    @Column(nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private SchoolCategory category;

    @Column(columnDefinition = "text")
    private String description;

    @Column(columnDefinition = "text")
    private String history;

    @Column
    private String city;

    @Column
    private String region;

    @Column
    private String address;

    @Column
    private Double latitude;

    @Column
    private Double longitude;

    @Column(name = "tuition_fee")
    private BigDecimal tuitionFee;

    @Column(nullable = false)
    private String currency = "XAF";

    @Column
    private String website;

    @Column
    private String phone;

    @Column
    private String email;

    @Column(name = "cover_image_url")
    private String coverImageUrl;

    @Column(nullable = false)
    private boolean approved = true;

    @Column(name = "created_at", nullable = false)
    private OffsetDateTime createdAt = OffsetDateTime.now();

    @OneToMany(mappedBy = "school", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Program> programs = new ArrayList<>();

    @OneToMany(mappedBy = "school", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SchoolImage> images = new ArrayList<>();

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    public School(String name, SchoolCategory category) {
        this.name = name;
        this.category = category;
    }
}

File: backend/src/main/kotlin/com/schoolfinder/domain/School.kt
package com.schoolfinder.domain

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(name = "school")
class School(
    @Column(nullable = false)
    var name: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var category: SchoolCategory,

    @Column(columnDefinition = "text")
    var description: String? = null,

    @Column(columnDefinition = "text")
    var history: String? = null,

    @Column
    var city: String? = null,

    @Column
    var region: String? = null,

    @Column
    var address: String? = null,

    @Column
    var latitude: Double? = null,

    @Column
    var longitude: Double? = null,

    @Column(name = "tuition_fee")
    var tuitionFee: BigDecimal? = null,

    @Column(nullable = false)
    var currency: String = "XAF",

    @Column
    var website: String? = null,

    @Column
    var phone: String? = null,

    @Column
    var email: String? = null,

    @Column(name = "cover_image_url")
    var coverImageUrl: String? = null,

    @Column(nullable = false)
    var approved: Boolean = true,

    @Column(name = "created_at", nullable = false)
    var createdAt: OffsetDateTime = OffsetDateTime.now(),

    @OneToMany(mappedBy = "school", cascade = [CascadeType.ALL], orphanRemoval = true)
    var programs: MutableList<Program> = mutableListOf(),

    @OneToMany(mappedBy = "school", cascade = [CascadeType.ALL], orphanRemoval = true)
    var images: MutableList<SchoolImage> = mutableListOf(),

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
)
